newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -24,95 +20,31 @@ export type DataSource = Array<
       value: string, // The value to show on screen and to be selected
       onClick?: () => void, // If defined, will be called when the item is clicked. onChange/onChoose won't be called.
       renderIcon?: ?() => React.Element<typeof ListIcon | typeof SvgIcon>,
-    |}
->;
+    |};
+
+// type Option is defined in isolation to DataSource, since it needs to be
+// passed as an object for some funtions.
```

Comment:
Thanks for the explanation 👍 I think we can remove this comment as this will be clear for the reader that will look at the code and see that Option is being used. Still a good thing for you to be conscious of this :)